wolfssl/internal.h

```diff
@@ -2761,7 +2761,7 @@ struct WOLFSSL_BIO {
     int          wrIdx;         /* current index for write buffer */
     int          rdIdx;         /* current read index */
     int          readRq;        /* read request */
-    int          num;           /* socket num or length */
+    SOCKET_T     num;           /* socket num or length */
```

Comment:
Any reason we have to use num for socket?

Reply:
`num` is used for the size in some BIO's and the file descriptors too. Not sure if `SOCKET_T` is the correct type to use for that. I would remove `num` and create new members for each use case to make it easier to identify what needs fixing.

Reply:
after comparing notes with Juliusz, I've opted to refactor `.num`, and `.ptr` for completeness, as `union`s.  it looks very nice now.